docs/DeveloperGuide.md

```diff
@@ -1,34 +1,890 @@
+    4. If recipe exists, the `CookRecipeCommand#checkIfRecipeExists()` method will return the index of the recipe, else it will return a number that is bigger than the size of `recipelist`.
+    5. Next, it is to check if there are sufficient ingredients to be deducted from the ingredients' inventory to cater for the number of pax for the specific recipe by calling `CookRecipeCommand#checkForSufficientIngredients()` method.
+    6. If `CookRecipeCommand#checkForSufficientIngredients()` returns true, then `CookRecipeCommand#deductIngredients()` will be called to deduct the ingredients in the ingredients' inventory.
+    6. Lastly, a String called `feedbackToUser` will be returned to the user to inform the user of the outcome of the command.
+4. The details will then be printed onto the console using `Ui#showResultToUser(result)`.
+
+##### Design Considerations 
+
+#### 4.2.4. Delete all/ specific recipe(s)
+The deletion feature for specific recipes allows the user to delete recipes either by the name or index of the recipe. 
+
+![Delete Recipe Sequence Diagram](images/deleteRecipeSequenceDiagram.png)
+##### Implementation
+When the user attempts to delete the `Chicken Rice` recipe from Kitchen Helper, the `Kitchen Helper`, `Parser` and `DeleteRecipeCommand` class will be called upon. The following sequence of steps will then occur: 
+1. The user keyed in “deleterecipe /n `Chicken Rice`".
+    2. A `UI` object will be created and it will call `UI#getUserCommand()` method to take in the input that the user has keyed in. 
+    3. A `String` object will be returned and saved into the `userCommandInput` variable in `Kitchen Helper`. 
+    4. The variable `userCommandInput` is being parsed into the `Parser` class as an argument for this method `Parser#parseUserCommand()`.
+2. The command inserted by the user is being parsed into the `Parser` and a new `Command` object is being created. 
+    2. The variable `userCommandInput` will be identified as `deleterecipe` in the `Parser#parseUserCommand()`.The `Parser#prepareDeleteRecipe()` is being called to prepare the `userCommandInput` string to create a `DeleteRecipeCommand` object.
+3. The command is now being executed.
+    2. The `DeleteRecipeCommand#execute()` will be called.
+    3. As this is a deletion by recipe name, the `recipeIndex` variable is set as null. As the variable is null, `DeleteRecipeCommand#deleteRecipeByName()` will be called.
+    4. Next, the `DeleteRecipeCommand#getRecipeIndex()` to get the index based on the recipe name that the user has inputted. With the given index, `DeleteRecipeCommand#deleteRecipe()` will be called to delete the recipe. 
+    5. Lastly, a String called `feedbackToUser` will be returned to the user to inform the user of the outcome of the command. 
+4. The details will then be printed onto the console using `Ui#showResultToUser(result)`.
+
+
+##### Design Considerations
+Aspect: How is the `DeleteRecipeCommand` initialise. <br>
```

Comment:
Would you consider phrasing the aspect as a statement than a question? 
For example: "Initialization of the `DeleteRecipeCommand`."